src/main/java/tool/arenasizer/ArenaInfoPanel.java

```diff
@@ -0,0 +1,174 @@
+
+        futureCapacityPanel.label2.setText(Optional.ofNullable(stadium.getExpansionDate())
+                .map(expansionDate -> nextDay(expansionDate).toLocaleDate())
+                .orElse(EMPTY));
+
+        setTranslation();
+    }
+
+    private static HODateTime nextDay(HODateTime hoDateTime) {
+        var localDate = hoDateTime.getLocalDateTime().toLocalDate();
+        var nextDay = localDate.plusDays(1);
+        var instant = Instant.from(nextDay.atStartOfDay(HODateTime.DEFAULT_TIMEZONE));
+        return new HODateTime(instant);
+    }
+
+    private void setTranslation() {
+        currentCapacityPanel.setTranslation();
+        currentCapacityPanel.setTitle(getLanguageString("ArenaInfoPanel.current"));
+        currentCapacityPanel.label1.setText(getLanguageString("ArenaInfoPanel.last_improvement"));
+
+        expandedCapacityPanel.setTranslation();
+        expandedCapacityPanel.setTitle(getLanguageString("ArenaInfoPanel.expansion"));
+        expandedCapacityPanel.labelPercentage.setText(getLanguageString("ArenaInfoPanel.change_in_percent"));
+        expandedCapacityPanel.label1.setText(getLanguageString("ArenaInfoPanel.completion"));
+
+        futureCapacityPanel.setTranslation();
+        futureCapacityPanel.setTitle(getLanguageString("ArenaInfoPanel.future"));
+        futureCapacityPanel.label1.setText(getLanguageString("ArenaInfoPanel.available"));
+    }
+
+    private void updateExpansionFinishedLabel() {
+        final JLabel labelExpansionFinished = expandedCapacityPanel.label3;
+        final Stadium stadium = HOVerwaltung.instance().getModel().getStadium();
+        Optional.ofNullable(stadium).map(Stadium::getExpansionDate).ifPresentOrElse(expansionDate -> {
+                    final String text = toDurationString(expansionDate);
+                    final String toolTipText = String.format(getLanguageString("ArenaInfoPanel.finished_in_n_days_format"), calculateDaysToNow(expansionDate, 1));
+                    labelExpansionFinished.setText(text);
+                    labelExpansionFinished.setToolTipText(toolTipText);
+                },
+                () -> {
+                    labelExpansionFinished.setToolTipText(null);
+                    labelExpansionFinished.setText(EMPTY);
+                }
+        );
+    }
+
+    private static String getLanguageString(String key) {
+        return HOVerwaltung.instance().getLanguageString(key);
+    }
+
+    private static BigDecimal calculateDaysToNow(HODateTime hoDateTime, int scale) {
+        return BigDecimal.valueOf(hoDateTime.getInstant().toEpochMilli() - HODateTime.now().getInstant().toEpochMilli())
+                .divide(BigDecimal.valueOf(1000L * 86400L), scale, RoundingMode.HALF_DOWN);
+    }
+
+    private static String toDurationString(HODateTime hoDateTime) {
```

Comment:
This method has two concerns that should probably be separated, the conversion of time till now to day/hour/min/sec (which should probably be pushed to `HODateTime`), and the building of the string to display in the panel.

Reply:
Review Finding 2:

Added `HumanDuration`